CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1886,7 +1886,7 @@ public function setFormAmountFields($priceSetID) {
       // contribution amount to be null, so that it will not show
       // contribution amount same as membership amount.
       //@todo - merge with section above
-      if ($this->_membershipBlock['is_separate_payment']
+      if (!empty($this->_membershipBlock['is_separate_payment'])
```

Comment:
Same question - why no enotices already?

Reply:
Seems like it was only added with php7.4 https://www.php.net/manual/en/migration74.incompatible.php 

"Array-style access of non-arrays ¶
Trying to use values of type null, bool, int, float or resource as an array (such as $null["key"]) will now generate a notice."